src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-requests/request-item/request-item.component.ts

```diff
@@ -19,4 +24,20 @@ export class RequestItemComponent {
   public decline(id: number) {
     this.declineEvent.emit(id);
   }
+
+  public clickHandler(event: MouseEvent): void {
+    const target = event.target as HTMLElement;
+    if (target.tagName === 'button') {
+      return;
+    } else {
+      this.redirectToFriendPage();
+    }
+  }
```

Comment:
Can you explain the idea here?

Reply:
redirectToFriendPage() is linked to the whole card, but we have two buttons that must have another functionality for clicking that will be implementing in the future

Reply:
I think you need to attach event handler to avatar only. If user needs to be redirected when clicking on card anywhere outside of buttons, i think you can use either `(click)="clickHandler($event); $event.stopPropagation()"` in template directly, or put `event.stopPropagation()` inside your `clickHandler` function.

Reply:
In the scope of this task, a decision is supposed to be made with a click on the card, not on the avatar. In addition, the similar functionality has another page with with cards. Maybe it’s better to clarify with an expert